content/blog/2023/2023-08-25-fairness-reweighing-preprocessor.md

```diff
@@ -0,0 +1,51 @@
++++
+author = "Žan Mervič"
+date = "2023-08-25"
+draft = false
+title = "Orange Fairness - Reweighing as a preprocessor"
+type = "blog"
+thumbImage = "/blog_img/2023/2023-08-25-fairness-reweighing-preprocessor-thumb.png"
+frontPageImage = "/blog_img/2023/2023-08-25-fairness-reweighing-preprocessor-thumb.png"
+blog = ["fairness", "reweighing"]
+shortExcerpt = "Expanding on the Orange fairness Reweighing widget: using it as a preprocessor and integrating new fairness scoring metrics."
+longExcerpt = "Diving deeper into the Orange fairness Reweighing widget, we explore its use as a preprocessor for models. Discover the new widgets and fairness scoring metrics; all illustrated using the German credit dataset, supplemented with visual insights through box plots."
++++
+
+In the [previous blog post]({{< ref "/blog/2023/2023-08-24-fairness-reweighing-dataset.md" >}}), we introduced the Orange fairness Reweighing widget and used it to reweigh a dataset. In this blog, we will explore another use case for the Reweighing widget: using it as a preprocessor for a specific model. Doing so will allow us to use the reweighing algorithm when evaluating model performance using cross-validation, random sampling, etc.
+
+### Weighted Logistic Regression and Combine Preprocessors:
+
+At the time of writing, there were limitations in using the Reweighing widget as a preprocessor. We introduced two new widgets to address this: Weighted Logistic Regression and Combine Preprocessors. The Weighted Logistic Regression widget is an adapted form of the Logistic Regression widget designed to handle weighted datasets. On the other hand, the Combine Preprocessors widget enables users to merge multiple preprocessors into one. This becomes handy when integrating preprocessors other than Reweighing with a model.
+
+### New scoring metrics in Test and Score and Predictions:
+
+With the fairness addon and widgets that come with it, we also introduced four new scoring metrics for the Test and Score and Predictions widgets. We use these metrics to evaluate the fairness of model predictions. The metrics are:
+
+- Disparate Impact (DI): Measures the ratio of favorable outcomes for an unprivileged group to that of the privileged group. An ideal value of 1.0 means the ratio of favorable outcomes is the same for both groups.
+  - If DI < 1.0: The privileged group receives favorable outcomes at a higher rate than the unprivileged group.
+  - If DI > 1.0: The privileged group receives favorable outcomes at a lower rate than the unprivileged group.
+- Statistical Parity Difference (SPD): This is very similar to disparate impact. Instead of the ratio, it measures the difference in favorable outcomes between the unprivileged and the privileged groups. An ideal value for this metric is 0.0, which means that the difference in favorable outcomes is the same for both groups.
```

Comment:
This is very similar to disparate impact. Instead of the ratio, it measures the difference in favorable outcomes between the unprivileged and the privileged groups. -> Similar to disparate impact, but measured the difference instead of the ratio. Ideally 0.